File: src/main/java/gargoyle/rpycg/ui/model/FULLNESS.java
package gargoyle.rpycg.ui.model;

import java.util.Arrays;
import java.util.Comparator;

public enum FULLNESS {
    NORMAL(0),
    ALMOST(10),
    FULL(12);
    private final int size;

    FULLNESS(int size) {
        this.size = size;
    }

    public static FULLNESS determineFullness(int size) {
        return Arrays.stream(values()).sorted(Comparator.comparingInt(o -> -o.size))
                .filter(fullness -> size >= fullness.size).findFirst().orElse(NORMAL);
    }

    public int getSize() {
        return size;
    }
}

File: app/src/main/kotlin/gargoyle/rpycg/ui/model/FULLNESS.kt
package gargoyle.rpycg.ui.model


enum class FULLNESS(val size: Int) {
    NORMAL(0), ALMOST(10), FULL(12);

    companion object {
        fun determineFullness(size: Int): FULLNESS =
            values()
                .sortedBy { -it.size }
                .firstOrNull { size >= it.size } ?: NORMAL
    }
}
